File: arrow.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func getArrowString(arrowSize int) string {
	arrow := ""
	for line := 0; line < arrowSize; line++ {
		arrow += strings.Repeat(" ", arrowSize-line)
		arrow += "*"
		if line != 0 {
			arrow += strings.Repeat(" ", 1+(line-1)*2)
			arrow += "*"
		}
		arrow += "\n"
	}
	arrow += strings.Repeat("*", arrowSize*2+1)
	arrow += "\n"
	tailLine := strings.Repeat(" ", arrowSize) + "*" + strings.Repeat(" ", arrowSize) + "\n"
	arrow += strings.Repeat(tailLine, arrowSize)
	return arrow
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("[" + os.Args[0] + "] Please enter one argument")
		return
	} else if len(os.Args) != 2 {
		fmt.Println("[" + os.Args[0] + "] Please enter ONLY one argument")
		return
	}
	arrowSize, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Println("First argument must be an integer")
		return
	}

	arrow := getArrowString(arrowSize)
	fmt.Print(arrow)
}

File: arrow.rs
use std::env;

fn print_arrow(size: usize) {

    let mut line = 0;
    let mut text: String;
    while line < size {
        text = " ".repeat(size-line);
        text.push_str("*");

        if line != 0 {
            text.push_str(&" ".repeat(1+(line-1)*2));
            text.push_str("*");
        }
        println!("{}", text);
        line = line+1;
    }
    text = "*".repeat(size*2+1);
    println!("{}", text);
    line = 0;
    while line < size {
        text = " ".repeat(size);
        text.push_str("*");
        println!("{}", text);
        line = line+1;
    }
}

fn main() {

    let args: Vec<String> = env::args().collect();

    match args.len() {
        
        1 => {
            println!("[{}] Please enter one argument", args[0]);
        },

        2 => {
            let result = args[1].parse::<usize>();
            match result {
                Ok(foo) => print_arrow(foo),
                Err(err) => println!("[{}]First argument must be an integer !", args[0]),
            }
        },

        x => {
            println!("[{}] Please enter ONLY one argument", args[0]);
        },
    }

}
